wolfcrypt/src/port/iotsafe/iotsafe.c

```diff
@@ -453,7 +453,7 @@ static int iotsafe_readfile(uint8_t *file_id, uint16_t file_id_sz,
         return ret;
     }
 
-    filesz_s = search_tlv(resp + 4, ret, 0x20);
```

Comment:
This change breaks the parser in the other demos. A well-formed response for the `get data - file` command starts with `C3` (cfr. IoT.05 section 3.9.4.1). The C3 tag in turn contains the File Information structure (sec. 3.10.4.2). The search_tlv funcion must skip the C3 tag and its length to parse its content (that's the reason for the 4-byte offset there). `0x20` is the tag for the file size (with fixed len 2).

Curious to know why you would need this change, and if the applet is supposed to return a file information structure without the C3 header.

Reply:
Note: `get data - file` operations fails with other applets when this change is applied

Reply:
good point, I will check on that part.

Reply:
I will revert the change here. You are right.